File: 4 Kyu/adding_big_numbers.js
function add (foo, bar) {
    let k = '', x = 0;

    foo = foo.split('');
    bar = bar.split('');

    while (foo.length || bar.length || x) {
        x += ~~foo.pop() + ~~bar.pop();
        k = x % 10 + k;
        x = x > 9;
    }
    return k;
}

File: 4 Kyu/Adding Big Numbers/C++/adding_big_numbers.cpp
#include <string>
#include <numeric>

std::string add(const std::string& foo, const std::string& bar) {
    std::string g;

    int k = 0;

    for (auto i = foo.rbegin(), j = bar.rbegin(); i != foo.rend() || j != bar.rend(); i++, j++) {
        int x = i != foo.rend() ? *i - '0' : (i = foo.rend() - 1, 0);
        int z = j != bar.rend() ? *i - '0' : (j = bar.rend() - 1, 0);

        g = std::to_string((x + z + k) % 10) + g;
        k = (x + z + k) / 10;
    }
    return k ? std::to_string(k) + g : g;
}
